### DISCUSSION TITLE
Exit message template

### DISCUSSION THREAD

**Discussion by 5011-5011** [nextgenhealthcare/connect#6071] [Q&A] [answered]
I receive a json message which I then convert into a hl7 message in a transformer source. The problem is that I have a script to handle the obx segments that are generated according to conditions. 
When I send the composed hl7 message an extra obx segment is generated without information, and it is the same that I incorporated in Outbound Message Template, when I remove it from Outbound Message Template the channel generates an error and does not process the message. 
Can someone guide me why this occurs? I attach images: 

![image](https://github.com/nextgenhealthcare/connect/assets/98954172/6a5c4b4e-14cc-40ec-8351-a17aff3671dd)

![image](https://github.com/nextgenhealthcare/connect/assets/98954172/4ac4888b-9366-4bf5-b2d4-78e6d7d92622)

**Comment by jonbartels** [comment, marked as answer]
The outbound template is used for like filling in a message. So if you have a sample message that you need to send, you can paste it into the outbound template. Then your transformers can use the drag-and-drop mappings to fill in the template. 

For the code you showed, what you could do instead of `createSegment` is just set `tmp['OBX'][0]['OBX.3'] = yourvalue` . If you know you will ALWAYS have 2 OBX segments you can build the template then fill it in. If you have a random number of OBX segments then keep using `createSegment` and remove OBX from the template

**Reply by 5011-5011** [reply]
Thanks for your help! The thing is that if they are random segments and removing the obx from the template does not process the step of the transformer, and I get the following error:

![image](https://github.com/nextgenhealthcare/connect/assets/98954172/cef747c1-7206-41d3-820a-5227722e4b6f)

**Reply by rjkroll** [reply]
You can't hardcode the Index values (i.e. [obx][1], [obx][2], etc) if you are not going to have 2 OBX segments in each message.   If you don't know how many you are going to have, you should be using a counter variable (i.e. i = 0 ->conditions/loop -> i++) as your index that you increment via a loop or some other condition.    That way, the index counter will also just try and create how many OBX's you need.

**Reply by jonbartels** [reply]
You can also just do 

```
var newObx = createSegment('OBX', tmp,1);
newObx['OBX.3']['OBX.3.1'] = 'hola';
newObx['OBX.4']['OBX.4.1'] = 'howdy';
```

`createSegment` returns the segment that was created.
